### PR TITLE
[FE-10] [FE-149] Bonuses section with hover

### PR DESCRIPTION
**Pull request by AnCichocka**
https://oops2000.atlassian.net/browse/SCRUM-10
https://oops2000.atlassian.net/browse/SCRUM-149

I didn't contain `description` in hover, because I don't know where to find it in hasura, do we have `description` similar field, @Szyntos ?

### CODE REVIEW

File: frontend/src/components/StudentProfile/cards/BonusesCard.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { Bonus } from "../../../hooks/StudentProfile/useBonusesCardData";
```
Comment (Szyntos) [inline frontend/src/components/StudentProfile/cards/BonusesCard.tsx:1]: Theoretically this should be an Award

In backend Award is the entity that gives the points (LEKARSTWO)
and Bonus is the result of that award, i.e. a triple (chestHistory, award, points)

IDK if it's worth the trouble tho

File: frontend/src/components/StudentProfile/table/cellContent/PointsCellContent.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import { Styles } from "../../../../utils/Styles";
+
+type PointsCellProps = {
+  points: Points;
+};
+
+export const PointsCellContent = ({ points }: PointsCellProps) => {
+  const bonusesPoints =
+    points.points.partialBonusType.length === 0
+      ? undefined
```
Comment (Szyntos) [inline frontend/src/components/StudentProfile/table/cellContent/PointsCellContent.tsx:12]: Isn't the inverse more readable?

```js
points.points.partialBonusType.length != 0
      ? points.points.partialBonusType.reduce(
          (total, bonus) => total + (bonus?.partialValue ?? 0)
          , 0)
        ? undefined
```

truly, idk

File: frontend/src/components/StudentProfile/table/cellContent/PointsCellContent.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  const purePoints = points.points.purePoints?.value
+    ? parseFloat(points.points.purePoints?.value)
+    : undefined;
+
+  const allPoints = (purePoints ?? 0) + (bonusesPoints ?? 0);
+
+  const getPointsDescriptionString = (): string | undefined => {
+    if (!bonusesPoints && !purePoints) {
+      return undefined;
```
Comment (Szyntos) [inline frontend/src/components/StudentProfile/table/cellContent/PointsCellContent.tsx:26]: Label in points tells you about the origin of the points, i.e. "Awarded for Lekarstwo"

File: frontend/src/hooks/StudentProfile/useBonusesCardData.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  });
+
+  const bonuses: Bonus[] =
+    data?.bonuses.map((bonus) => {
+      return {
+        award: {
+          id: bonus.award.awardId,
+          name: bonus.award.awardName,
+          value: parseFloat(bonus.award.awardValue),
+          imgId: bonus.award.imageFileId ?? undefined,
```
Comment (Szyntos) [inline frontend/src/hooks/StudentProfile/useBonusesCardData.tsx:34]: There is image/bonus/sample on back

i could make samples as a permanent data on the db and we could use them here

Comment (Szyntos) [reply frontend/src/hooks/StudentProfile/useBonusesCardData.tsx:34]: But this is a separate ticket

**Review by Szyntos — CHANGES_REQUESTED** [review]
Looks nice, but shouldn't the hover be applied also on the bonuses in the student's table?

image

here?

File: frontend/src/components/StudentProfile/cards/BonusesCard.tsx (hunk 1 above)
Comment (AnCichocka) [reply frontend/src/components/StudentProfile/cards/BonusesCard.tsx:1]: I don't think so, as long as I know what it's all about

File: frontend/src/hooks/StudentProfile/useBonusesCardData.tsx (hunk 4 above)
Comment (AnCichocka) [reply frontend/src/hooks/StudentProfile/useBonusesCardData.tsx:36]: Can you create jira ticket?

File: frontend/src/components/StudentProfile/table/cellContent/PointsCellContent.tsx (hunk 2 above)
Comment (AnCichocka) [reply frontend/src/components/StudentProfile/table/cellContent/PointsCellContent.tsx:12]: tbh, no difference, I can change that

**Review by Szyntos — APPROVED** [review]
mlgt

**Pull request merged**